File: Python/PyCharm/learning_python/topic1/loops.py
# While loop
a = 10
while a > 0:
    print(a)
    a -= 2
    if a == 2: break;
else:
    print("Broke at 2")

# For loop
for i in [1, 2, 4, 5]:
    n = 2
    if n == i:
        print("n is in the list");
        break

# Nested loop

for i in range(1, 11):
    for j in range(i):
        print(j + 1)
    print()

File: Languages/Scala/Learning/src/main/scala/basics/loops.scala
package basics

object loops {
  def main(args: Array[String]): Unit = {
    // for loop
    println("With for loop")
    for (i <- 1 to 10) {
      println(i)
    }

    val randomLetters = "ABCDE"
    for (letter <- 0 until randomLetters.length) {
      println(randomLetters(letter))
    }

    // Looping List
    val list = List(1, 2, 3, 4, 5)
    for (i <- list) {
      println(i)
    }

    // While loop
    var i = 0
    println("With while loop")
    while (i <= 10) {
      println(i)
      i += 1
    }

    // Loop with yield, and without needing brackets.
    val evenList = for {i <- 1 to 20 if (i % 2) == 0} yield i
    for (i <- evenList)
      println(i)

    println("J is nested of I")
    for (i <- 1 to 5; j <- 6 to 10) {
      println("I : " + i)
      println("J : " + j)
    }

  }
}
